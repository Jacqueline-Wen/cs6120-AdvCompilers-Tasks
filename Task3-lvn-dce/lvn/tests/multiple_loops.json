{
    "functions": [
        {
            "instrs": [
                {
                    "dest": "a",
                    "op": "const",
                    "type": "int",
                    "value": 1
                },
                {
                    "dest": "b",
                    "op": "const",
                    "type": "int",
                    "value": 2
                },
                {
                    "args": [
                        "a",
                        "b"
                    ],
                    "dest": "c",
                    "op": "add",
                    "type": "int"
                },
                {
                    "args": [
                        "b",
                        "a"
                    ],
                    "dest": "d",
                    "op": "add",
                    "type": "int"
                },
                {
                    "args": [
                        "d"
                    ],
                    "dest": "e",
                    "op": "id",
                    "type": "int"
                },
                {
                    "args": [
                        "e"
                    ],
                    "op": "print"
                }
            ],
            "name": "main"
        }
    ]
}