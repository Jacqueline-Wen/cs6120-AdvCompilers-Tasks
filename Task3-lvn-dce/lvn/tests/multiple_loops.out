{"functions":[{"instrs":[{"dest":"a","op":"const","type":"int","value":1},{"dest":"b","op":"const","type":"int","value":2},{"dest":"c","op":"const","type":"int","value":3},{"args":["c"],"dest":"d","op":"id","type":"int"},{"args":["c"],"dest":"e","op":"id","type":"int"},{"args":["c"],"op":"print"}],"name":"main"}]}
